### PR TITLE
add 'id' option to DxApi.whoami

### PR DESCRIPTION
**Pull request by jdidion**

### CODE REVIEW

File: api/src/main/scala/dx/api/DxApi.scala [hunk 1]
```diff
@@ -496,6 +496,11 @@ case class DxApi(version: String = "1.0.0", dxEnv: DXEnvironment = DXEnvironment
       case other =>
         throw new Exception(s"unexpected whoami result ${other}")
     }
+    if (id) {
```
Comment (r-i-v-a) [inline api/src/main/scala/dx/api/DxApi.scala:499]: I don't think this is needed. DXAPI.systemWhoami is 
```
   public static <T> T systemWhoami(Object inputObject, Class<T> outputClass, DXEnvironment env) {
        JsonNode input = mapper.valueToTree(inputObject);
        return DXJSON.safeTreeToValue(
                new DXHTTPRequest(env).request("/system/whoami", input, RetryStrategy.SAFE_TO_RETRY),
                outputClass);
    }
```

Comment (r-i-v-a) [reply api/src/main/scala/dx/api/DxApi.scala:499]: Which means it's already consistent with the underlying API call dxpy uses 
http://autodoc.dnanexus.com/bindings/python/current/dxpy_api.html?highlight=whoami#dxpy.api.system_whoami
https://documentation.dnanexus.com/developer/api/system-methods#api-method-system-whoami

Comment (r-i-v-a) [reply api/src/main/scala/dx/api/DxApi.scala:499]: You can delete your folders and re-run tests to confirm this, but I am now seeing folder names builds/user-rnathans and unit_tests/user-rnathans created for me.

Comment (r-i-v-a) [reply api/src/main/scala/dx/api/DxApi.scala:499]: With this update, the clean_build script removes user-xxxx folders. The removed lines, and probably this API call, would generate user-user-xxxx. 
https://github.com/dnanexus/dxCompiler/pull/113/files#diff-1469ceedc853f6160d183753eb11a262a8cab0848c9c1d1f355b3a5f63750646L14

**Review by r-i-v-a — CHANGES_REQUESTED** [review]
Not needed. It turns out the Python and Scala APIs are consistent already.

File: api/src/main/scala/dx/api/DxApi.scala (hunk 1 above)
Comment (jdidion) [reply api/src/main/scala/dx/api/DxApi.scala:499]: ok thanks

**Pull request closed without merging**